src/contentChunk.js

```diff
@@ -44,20 +54,78 @@ exports.extractStringPayloadIntoChunks = (payload) => {
     if (!isValidGridFsPayload(payload)) {
       return reject(new Error('payload not in the correct format, expecting a string, Buffer, ArrayBuffer, Array, or Array-like Object'))
     }
-    
-    const bucket = new mongodb.GridFSBucket(connectionDefault.client.db())
-    const stream = bucket.openUploadStream()
 
-    stream.on('error', (err) => {
-      return reject(err)
-    })
+    const bucket = getGridFSBucket()
+    const uploadStream = bucket.openUploadStream()
+
+    uploadStream.on('error', reject)
     .on('finish', (doc) => {
       if (!doc) {
         return reject(new Error('GridFS create failed'))
       }
 
-      return resolve(doc._id)
+      resolve(doc._id)
     })
-    stream.end(payload)
+    uploadStream.end(payload)
+  })
+}
+
+exports.removeBodyById = (id) => {
+  return new Promise(async (resolve, reject) => {
+    if (!id) {
+      return reject(new Error('No ID supplied when trying to remove chucked body'))
+    }
+
+    try {
+      const bucket = getGridFSBucket()
+      const result = await bucket.delete(id)
+      resolve(result)
+    } catch (err) {
+      reject(err)
+    }    
+  })
+}
+
+export const retrievePayload = fileId => {
+  return new Promise((resolve, reject) => {
+    if (!fileId) {
+      return reject(new Error(`Payload id not supplied`))
+    }
+
+    const bucket = getGridFSBucket()
+    const chunks = []
+
+    bucket.openDownloadStream(fileId)
+      .on('error', err => reject(err))
+      .on('data', chunk => chunks.push(chunk))
+      .on('end', () => resolve(Buffer.concat(chunks).toString()))
+  })
+}
+
+export const addBodiesToTransactions = async (transactions) => {
+  if(!transactions ||
+    transactions.length < 1
+  ) {
+    return []
+  }
+
+  return await Promise.all(transactions.map(transaction => filterPayloadType(transaction)))
+}
+
+const filterPayloadType = (transaction) => {
+  return new Promise(async (resolve, reject) => {
```

Comment:
There aren't any error cases for this function. It always resolves what was passed in as a parameter, whether that is a transaction, null, undefined, etc. 
We could maybe refactor to make that clearer?

Reply:
It won't resolve if the retrievePayload throws an error.

I tested it now and that reject case is needed.

https://github.com/jembi/openhim-core-js/pull/1026/files#diff-6b387023209fd79314641ba533f5ce23R89

Reply:
Done in ad827ed

Reply:
Im wondering if this shouldnt be handled father up the chain, otherwise we need to be adding try/catch for everytime this is being used which wont be ideal

Reply:
This is an actual issue that needs to be handled correctly. Will add the proper error handling :+1: